x-pack/solutions/observability/packages/utils_server/es/storage/types.ts

```diff
@@ -91,7 +91,7 @@ type PrimitiveOf<TProperty extends StorageMappingProperty> = {
   float: number;
   object: TProperty extends { properties: Record<string, StorageMappingProperty> }
     ? {
-        [key in keyof TProperty['properties']]: StorageFieldTypeOf<TProperty['properties'][key]>;
+        [key in keyof TProperty['properties']]?: StorageFieldTypeOf<TProperty['properties'][key]>;
```

Comment:
> [!NOTE]
> This is the only way I could get the `ingest` and `grouped` to work together in the same `IndexStorgageAdapter`

Reply:
not gonna lie, this nerd-snipped me pretty hard, and I spent half an hour to come up with a better solution and failed.